src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -83,6 +84,22 @@ public function get(ProductId $productId): Product
         return $product;
     }
 
+    /**
+     * @param Product $product
+     *
+     * @return ProductId
+     *
+     * @throws CannotAddProductException
+     */
+    public function add(Product $product): ProductId
```

Comment:
I hesitate, I wonder if you should instead have a `create` method with only the needed parameters
Instead of prefilling the object in the handler, it would give a single easy endpoint to create Product (usable somewhere else)
I'm just worried if it's not already too much domain logic for a Repository, but we're only talking about one default value (`active`) the other three would be parameters: `$isVirtual` `$localizedNames` `$defaultCategoryId`

Reply:
According to the [Repository Design Pattern](https://medium.com/@pererikbergman/repository-design-pattern-e28c0f3e4a30) the constructed object must be given to the repository to be saved 🤔

Reply:
I've seen other Repositories where there is such a thing (like `ProductTagRepository::add()` I believe.

I think in "pure" DDD the Entity constructor contains the creation logic. As we dont have an equivalent, it's fine to have it in the Repository.